pyriksdagen/args.py

```diff
@@ -0,0 +1,163 @@
+    parser.add_argument("-e", "--end",
+                        type=int,
+                        default=None,
+                        help="end year. If -e is set, -s must also be set. If -s and -e are explicitly set, all protocols in that range are processed. If -s and -e are unset and neither -p nor -l are set, the script will process all records from 1867 until the current year.")
+    parser.add_argument("-y", "--parliament-year",
+                        type=int,
+                        default=None,
+                        nargs='*',
+                        help="Parliament year, e.g. 1971 or 198384")
+    parser.add_argument("-R", "--records-folder",
+                        type=str,
+                        default=None,
+                        help="(optional) Path to records folder, defaults to environment var or `data/`")
+    parser.add_argument("-r", "--records",
+                        type=str,
+                        default=None,
+                        nargs='*',
+                        help="operate on a single record or list of records. Set the path from ./ -- this option doesn't cooperate with `-R`. -r takes presidence over -l.")
+    parser.add_argument("-l", "--from-list",
+                        type=str,
+                        default=None,
+                        help="operate on a list of records from a file. Set the path from ./ -- this option doesn't cooperate with `-R`.")
+    return parser
+
+
+
+
+def record_args(args):
+    """
+    takes an argparse namespace object for working with riksdagen records and imputes standard stuff
+    args:
+        args: args
+    returns:
+        args
+    """
+    def _iterate_protocols(s, e, data_location):
+        records = sorted(list(protocol_iterators(data_location, start=s, end=e)))
+        return records
+
+    if (args.start is None or args.end is None) and args.start != args.end:
+        print("\n\nWARNING: Set -s and -e together or neither.\n\n")
+        sys.exit()
+
+    if args.records_folder is None:
+        args.records_folder = get_data_location("records")
+
+    if args.start is not None:
+        args.records = _iterate_protocols(args.start, args.end, args.records_folder)
+        print(args.records)
+    else:
+        if args.records is not None and len(args.records) != 0:
+            pass
+        elif args.from_list is not None:
+            with open(args.from_list, 'r') as inf:
+                lines = inf.readlines()
+                args.protocol = [_.strip() for _ in lines if _.strip() != '']
+        elif args.parliament_year is not None and len(args.parliament_year) != 0:
+            args.records = []
+            for py in args.parliament_year:
+                args.records.extend(glob(f"{args.records_folder}/{py}/*.xml"))
+        else:
+            args.records = _iterate_protocols(1867, int(datetime.now().strftime('%Y')), args.records_folder)
```

Comment:
AFAIK protocol_iterators has a reasonable default value of returning all protocols

Reply:
```py
protocol_iterators(corpus_root=path, document_type="prot")
```

Reply:
I'll work on this a bit -- I think protocol_iterators needs a bit of help since the move from westac structure to swerik structure.

Right, re line 90!otherwise the `for py in parliament_year` part hasn't been used with the protocols yet -- currently e.g. if we do `-e 2023` we get all the stuff in the calendar year 2023, but If I want 202223, but not 202324, there's no easy way to do that with the args. This strategy works nicely with IP debates and motions so far.

Reply:
@ninpnin `protocol_iterators` hadn't been updated with the post-westac corpus structure changes and document_type didn't work as expected. I've added a new function `corpus_iterator`, which does almost the same as `protocol_iterators`, except that it actually accounts for the doctype in the current corpus structure. `protocol_iterators` gets a deprecation warning, but still works by calling the new function so as not to break compatibility. 

Similarly `write_protocol` and `parse_protocol` get replaced by `write_tei` and `parse_tei`.

`args.impute_args` works better with the `corpus_iterator` defaults (per your review) and priorities for how the function deals with set/unset args are more transparent in the help strings.